contracts/mapping-token/contracts/v2/Backing.sol

```diff
@@ -0,0 +1,249 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import "@zeppelin-solidity-4.4.0/contracts/proxy/utils/Initializable.sol";
+import "@zeppelin-solidity-4.4.0/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@darwinia/contracts-bridge/contracts/interfaces/ICrossChainFilter.sol";
+import "@darwinia/contracts-bridge/contracts/interfaces/IOutboundLane.sol";
+import "@darwinia/contracts-utils/contracts/DailyLimit.sol";
+import "@darwinia/contracts-utils/contracts/Ownable.sol";
+import "@darwinia/contracts-utils/contracts/Pausable.sol";
+import "../interfaces/IBacking.sol";
+import "../interfaces/IERC20.sol";
+import "../interfaces/IMessageVerifier.sol";
+import "../interfaces/IMappingTokenFactory.sol";
+
+contract Backing is Initializable, Ownable, DailyLimit, ICrossChainFilter, IBacking, Pausable {
+    struct RegisterInfo {
+        address token;
+        uint32 bridgedChainPosition;
+        address mappingTokenFactory;
+    }
+    struct LockedInfo {
+        address token;
+        address sender;
+        uint256 amount;
+    }
+    uint32 public constant NATIVE_TOKEN_TYPE = 0;
+    uint32 public constant ERC20_TOKEN_TYPE = 1;
+    string public thisChainName;
+
+    // bridge channel
+    mapping(bytes32 => address) public inboundLanes;
+    mapping(bytes32 => address) public outboundLanes;
+
+    // tokenAddress => remoteChainId => mappingTokenFactory
+    mapping(address => mapping(uint32 => address)) public tokens;
+
+    // (messageId => RegisterInfo)
+    mapping(bytes32 => RegisterInfo) public registerMessages;
+    // (messageId => lockedInfo)
+    mapping(bytes32 => LockedInfo) public lockMessages;
+
+    event NewInBoundLaneAdded(address backingAddress, address inboundLane);
+    event NewOutBoundLaneAdded(address outboundLane);
+    event NewErc20TokenRegistered(bytes32 messageId, uint32 bridgedChainPosition, uint32 bridgedLanePosition, address mappingTokenFactory, address token);
+    event TokenLocked(bytes32 messageId, uint64 nonce, uint32 bridgedChainPosition, uint32 bridgedLanePosition, address token, address recipient, uint256 amount);
+    event TokenLockFinished(bytes32 messageId, uint64 nonce, bool result);
+    event TokenRegisterFinished(bytes32 messageId, uint64 nonce, bool result);
+    event TokenUnlocked(uint32 bridgedChainPosition, uint32 bridgedLanePosition, address mappingTokenFactory, address token, address recipient, uint256 amount);
+
+    modifier onlyInBoundLane(uint32 bridgedChainPosition, uint32 bridgedLanePosition, address mappingTokenFactoryAddress) {
+        bytes32 remoteId = keccak256(abi.encodePacked(bridgedChainPosition, bridgedLanePosition, mappingTokenFactoryAddress));
```

Comment:
Collision might be not possible, but still prefer to have stronger verify here(e.g. uses extra mappingTokenFactoryAddress sender checks).